types/lodash/lodash-tests.ts

```diff
@@ -5362,14 +5372,14 @@ fp.now(); // $ExpectType number
     _.get({ a: false }, "a"); // $ExpectType boolean
     _.get(objectWithOptionalField, "a"); // $ExpectType boolean | undefined
     _.get(objectWithOptionalField, "a" as string); // $ExpectType any
-    _.get({ a: arrayOfNumbers }, 'a.0'); // $ExpectType number
-    _.get({ a: arrayOfNumbers }, 'a[0]'); // $ExpectType number
-    _.get({ a: arrayOfNumbers }, `a[${anyNumber}]`); // $ExpectType number
+    _.get({ a: arrayOfNumbers }, 'a.0'); // $ExpectType number | undefined
+    _.get({ a: arrayOfNumbers }, 'a[0]'); // $ExpectType number | undefined
+    _.get({ a: arrayOfNumbers }, `a[${anyNumber}]`); // $ExpectType number | undefined
     _.get({ a: tupleOfNumbers }, 'a.0'); // $ExpectType 1
     _.get({ a: tupleOfNumbers }, 'a[0]'); // $ExpectType 1
     _.get({ a: tupleOfNumbers }, 'a[1]'); // $ExpectType undefined
-    _.get({ a: tupleOfNumbers }, `a[${anyNumber}]`); // $ExpectType undefined
-    _.get({ a: dictionary }, 'a.b'); // $ExpectType string
+    _.get({ a: tupleOfNumbers }, `a[${anyNumber}]`); // $ExpectType 1 | undefined
+    _.get({ a: dictionary }, 'a.b'); // $ExpectType string | undefined
     _.get(maybeObject, 'a.b'); // $ExpectType string | undefined
```

Comment:
I'd like to see some more tests for `_.get` itself, using multiple levels of dots and square brackets. The unit tests you added below are OK, but they don't prove that the system is working correctly as a whole.

Reply:
Added such tests.